sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/AsyncDocumentClient.java

```diff
@@ -242,6 +240,14 @@ public TokenResolver getTokenResolver() {
         public void setTokenResolver(TokenResolver tokenResolver) {
             this.tokenResolver = tokenResolver;
         }
+
+        public CosmosKeyCredential getCosmosKeyCredential() {
+            return cosmosKeyCredential;
+        }
+
+        public void setCosmosKeyCredential(CosmosKeyCredential cosmosKeyCredential) {
```

Comment:
I see,   withCosmosKeyCredential above, 
this should be consistent with it.

Reply:
Now when I am thinking about this, the setter shouldn't even be there, will remove it.